src/test/java/runge/memetests/MemeControllerTest.java

```diff
@@ -0,0 +1,120 @@
+package runge.memetests;
+
+import io.reactivex.rxjava3.core.Observable;
+import io.reactivex.rxjava3.plugins.RxJavaPlugins;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import org.junit.jupiter.api.Test;
+import runge.meme.*;
+
+import javax.swing.*;
+
+import java.awt.*;
+
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.verify;
+
+public class MemeControllerTest
+{
+
+    static {
+        //makes it so that our service returns right away
+        RxJavaPlugins.setIoSchedulerHandler(scheduler -> Schedulers.trampoline());
+        RxJavaPlugins.setNewThreadSchedulerHandler(scheduler -> Schedulers.trampoline());
+    }
+
+    @Test
+    public void addAndUpdateImage()
+    {
+        //given
+        ImageService service = mock();
+        DefaultListModel<String> listModel = mock();
+        JLabel imageLabel = mock();
+        MemeController controller = new MemeController(service, listModel, imageLabel);
+        MemeList memeList = mock();
+        Observable<MemeList> observable = Observable.just(memeList);
+        doReturn(observable).when(service).getMemeImage();
+
+
+        //when
+        controller.addAndUpdateImage("Evil Kermit");
+
+
+        //then
+        verify(service).getMemeImage();
+
+    }
+
+    @Test
+    public void updateText()
+    {
+        //given
+        ImageService service = mock();
+        DefaultListModel<String> listModel = mock();
+        JLabel imageLabel = mock();
+        MemeController controller = new MemeController(service, listModel, imageLabel);
+        MemeList memeList = mock();
+        Observable<MemeList> observable = Observable.just(memeList);
+        doReturn(observable).when(service).getMemeImage();
+        JTextField upperText = mock();
+        JTextField lowerText = mock();
+
+        //when
+        upperText.setText("Hello");
+        lowerText.setText("World");
```

Comment:
These should not be here.
You are essentially calling the methods that you are then testing for on lines 67-68

Reply:
I have no idea how to write these tests